lib/account.js

```diff
@@ -27,32 +56,94 @@ module.exports = function(config) {
+      });
+
+      loginRequest.end(JSON.stringify({
+        password: request.payload.password,
+        uid: request.payload.uid,
+        user: {}
+      }), 'utf8');
+    },
+    requestReset: function(request, callback) {
+      var appURLObj = url.parse(config.uri + '/reset-password', true);
+      appURLObj.query = request.payload.oauth;
+
+      var resetRequest = hyperquest({
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'x-ratelimit-ip': getIPAddress(request)
+        },
+        uri: loginAPI + '/api/v2/user/request-reset-code'
+      });
+
+      resetRequest.on('error', callback);
 
-        message.on('data', function(c) {
-          bodyParts.push(c);
-          bytes += c.length;
+      resetRequest.on('response', function(message) {
+        parseMessage(message, function(err, json) {
+          callback(err, json);
         });
+      });
 
-        message.on('end', function() {
-          var body = Buffer.concat(bodyParts, bytes).toString('utf8');
-          var json;
+      resetRequest.end(JSON.stringify({
+        uid: request.payload.uid,
+        appURL: url.format(appURLObj)
+      }), 'utf8');
+    },
+    resetPassword: function(request, callback) {
+      var resetRequest = hyperquest({
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'x-ratelimit-ip': getIPAddress(request)
+        },
+        uri: loginAPI + '/api/v2/user/reset-password'
+      });
 
-          try {
-            json = JSON.parse(body);
-          } catch (ex) {
-            return callback(ex);
-          }
+      resetRequest.on('error', callback);
 
-          callback(null, json.user);
+      resetRequest.on('response', function(message) {
+        parseMessage(message, function(err, json) {
+          callback(err, json);
         });
       });
 
-      loginRequest.end(JSON.stringify({
-        password: request.payload.password,
+      resetRequest.end(JSON.stringify({
         uid: request.payload.uid,
-        user: {}
+        resetCode: request.payload.resetCode,
+        newPassword: request.payload.password
+      }), 'utf8');
+    },
+    createUser: function(request, callback) {
+      var resetRequest = hyperquest({
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'x-ratelimit-ip': getIPAddress(request)
+        },
+        uri: loginAPI + '/api/v2/user/create'
+      });
+
+      resetRequest.on('error', callback);
+
+      resetRequest.on('response', function(message) {
+        parseMessage(message, function(err, json) {
+          callback(err, json);
+        });
+      });
+
+      resetRequest.end(JSON.stringify({
+        user: {
+          email: request.payload.email,
+          username: request.payload.username,
+          password: request.payload.password,
+          mailingList: request.payload.feedback,
+          prefLocale: request.payload.lang
```

Comment:
are we expecting this to be pass in somewhere from the client? or we are just hardcoding the value for now? Since the UI doesn't support that currently.